src/main/java/uk/gov/pay/publicauth/model/TokenIdentifier.java

```diff
@@ -0,0 +1,6 @@
+package uk.gov.pay.publicauth.model;
+
+public interface TokenIdentifier {
```

Comment:
I’m not sure this interface is pulling its weight. It’s really only used to allow a single method — `AuthDao.revokeSingleToken(String, TokenIdentifier)` — to be polymorphic and `getColumnName()` means that our (otherwise lovely) domain objects `TokenHash` and `TokenLink` have to know the names of the database columns they’re stored in.

I think it might just be simpler to have `PublicAuthResource.revokeSingleToken(String, JsonNode)` determine whether it has a token link or a token hash and then call different variants of the `revokeSingleToken` method in `AuthDao` (which can of course delegate much of their common functionality to a shared private method).

Reply:
I originally did it like you proposed and was a bit underwhelmed by the results. Let me try to give it another go and see 👍

Reply:
That also will get rid of ```       TokenIdentifier token = Optional.ofNullable(payload.get(TOKEN_HASH_FIELD))
                .map(t -> (TokenIdentifier) TokenHash.of(t.asText()))
                .orElseGet(() -> TokenLink.of(payload.get(TOKEN_LINK_FIELD).asText()));``` of which I am very proud. 💀 🗡 😢